include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,382 @@
+				return false;
+
+			return true;
+		}
+
+		static inline bool execute(state_type* state)
+		{
+			if (!validate(state))
+				return false;
+
+			// load all the state
+			const auto* const inImg = state->inImage;
+			auto* const outImg = state->outImage;
+			const ICPUImage::SCreationParams& inParams = inImg->getCreationParameters();
+			const ICPUImage::SCreationParams& outParams = outImg->getCreationParameters();
+			const auto inFormat = inParams.format;
+			const auto outFormat = outParams.format;
+			const auto inBlockDims = asset::getBlockDimensions(inFormat);
+			const auto outBlockDims = asset::getBlockDimensions(outFormat);
+			const auto texelByteSize = asset::getTexelOrBlockBytesize(outFormat);
+			const auto* const inData = reinterpret_cast<const uint8_t*>(inImg->getBuffer()->getPointer());
+			auto* const outData = reinterpret_cast<uint8_t*>(outImg->getBuffer()->getPointer());
+			const core::SRange<const IImage::SBufferCopy> outRegions = outImg->getRegions(state->outMipLevel);
+
+			const auto inMipLevel = state->inMipLevel;
+			const auto outMipLevel = state->outMipLevel;
+			const auto inBaseLayer = state->inBaseLayer;
+			const auto outBaseLayer = state->outBaseLayer;
+			const auto layerCount = state->inLayerCount;
+			assert(layerCount == state->outLayerCount); // validation bug?
+
+			const auto inOffset = state->inOffset;
+			const auto outOffset = state->outOffset;
+			const auto inExtent = state->inExtent;
+			const auto outExtent = state->outExtent;
+
+			const auto inOffsetBaseLayer = state->inOffsetBaseLayer;
+			const auto outOffsetBaseLayer = state->outOffsetBaseLayer;
+			const auto inExtentLayerCount = state->inExtentLayerCount;
+			const auto outExtentLayerCount = state->outExtentLayerCount;
+			const auto inLimit = inOffsetBaseLayer + inExtentLayerCount;
+			const auto outLimit = outOffsetBaseLayer + outExtentLayerCount;
+
+			CMatchedSizeInOutImageFilterCommon::CommonExecuteData commonExecuteData =
+			{
+				inImg,
+				outImg,
+				inParams,
+				outParams,
+				inParams.format,
+				outParams.format,
+				getTexelOrBlockBytesize(inParams.format),
+				getTexelOrBlockBytesize(outParams.format),
+				reinterpret_cast<const uint8_t*>(inImg->getBuffer()->getPointer()),
+				reinterpret_cast<uint8_t*>(outImg->getBuffer()->getPointer()),
+				inImg->getRegions(state->inMipLevel),
+				outRegions,
+				outRegions.begin(), {}, {}
+			};
+
+			auto allocateCache = [](core::smart_refctd_ptr<asset::ICPUBuffer> cache, size_t size)
+			{
+				auto newPointer = reinterpret_cast<uint8_t*>(_IRR_ALIGNED_MALLOC(size, _IRR_SIMD_ALIGNMENT));
+				cache = core::make_smart_refctd_ptr<asset::CCustomAllocatorCPUBuffer<core::null_allocator<uint8_t>>>(size, newPointer, core::adopt_memory);
+			};
+
+			allocateCache(state->rowSumCache, outExtent.X * texelByteSize);
+			allocateCache(state->columnSumCache, outExtent.Y * texelByteSize);
```

Comment:
you ask for scratch memory so that you do not need to allocate!

Reply:
but it isn't an array of double that I use as scratch
it's for caching purposes, for storing max values for the pixel sum likely, the sum will be changing but it's cache size will be always the sam - extent of y and x,

so I use (will use) it for caching to speed up performing time and scratch memory that is allocated by the user to finally hold all the precomputed sum values

Reply:
all memory yyou need to use need to come from scratch memory,compute yourself some offsets and deal with everything via `reinterpret_cast`

Reply:
done likely